MNIST_skeleton.py

```diff
@@ -15,11 +15,14 @@
                    ])),
     batch_size=64, shuffle=True)
 
-
-# TODO: implement squash function
 def squash(input):
-    return input
-
+    """
+    Squashing function for a tensor.
+    :param input: torch.Tensor
+    """
+    norm = input.norm()
+    squared_norm = norm.pow(2)
+    return (squared_norm/(1+squared_norm))*(input/norm)
```

Comment:
The norm of a vector of size 0 is 0, if at least one of the vectors is of size 0, the second term of the equation will be a division by zero. We could add a small epsilon value to the denominator, however all other implementations seem to implement a much more convoluted solution.

Reply:
Yeah, that is true. I think it would actually be best practice. 

It would practically never happen to have a vector norm equal to 0, but still. Better do it right.

Reply:
So, from my perspective there's two things we can do:
1. Add an epsilon value to the denominator of the second term in the equation.
2. Leave it as is and write some tests in case division by zero happens. That way we'll know what went wrong and not waste time looking for the problem. Although it will be very obvious if it happens. Just tried it out and it results in a an `inf.` value for every element in the vector in the numerator.